### ISSUE TITLE
Characters should roll after role selection

### ISSUE THREAD

**Issue by metalgearsloth** [space-wizards/space-station-14#29705]
We bikeshedded this during loadouts and it came up again with nukies.

tl;dr is that job selection should be entirely independent of character and rolled first (same with antags). Once this is done then you should be able to pick which character you want to play.

**Comment by Doritodude77** [comment]
Completely agree, the good idea of an "antag only" character for what your regular character might be RP opposed to is inexecutable without this as you can't predict what you're going to get

**Comment by Kayek14** [comment]
will this give people the option to use clown/mime names? If so, then yes i agree!

**Comment by metalgearsloth** [comment]
> will this give people the option to use clown/mime names? If so, then yes i agree! But if this allows players to have specific names for antag roll then i disagree

That just requires adding a field to the RoleLoadout entry for that role which is separate and much easier.

**Comment by TurboTrackerss14** [comment]
So the concern I have with this is it leading to metagaming and it leaning into "allowed metagaming"

Say I play Daniel Smithson, but Danny Smithson whenever I roll antagonist. 
I could have my characters eyes be blood-red whenever I get antagonist.

While I can understand the character progression and interesting roleplay stance on this, I fear that the risks out way the benefits if I am reading the tldr correctly..

**Comment by metalgearsloth** [comment]
Okay I'm inclined to think:
- Roll antags.
- Notify player they have an antag roll. If it's exclusive (e.g. nukies, ninja) then don't let them select a character and they load in. At this point we either force them into a job that can play as that antag or just reroll it if they pick one that can't play it.
- If not they can then select one of their compatible characters for the job.

If you want to keep latejoins distinct then notify them like 10-15 seconds before the round starts so they have time to select.

We would also remove job-specific entities and just add them as potential characters, so borgs + AI + whatever else comes up. This would also make handling names a bit cleaner.
The main issue is they don't need markings etc. data, so ideally:
- HumanoidCharacterProfile -> SpeciesData, where SpeciesData between borgs / AI / normal mobs is different and shows different UI options.

**Comment by FairlySadPanda** [comment]
Apols for ideasguying here but this feels like a bit of a larger discussion about start-of-round flow, it's strongly adjacent to the topic of roundstart scenarios (i.e. where do characters spawn at roundstart, what condition are they in, are you a new shift just getting started, are you all at departures, is it supposed to be in media res, are you all staggering awake after a boozy party, etc)

In particular it's this engineering of a flow of: round start timer begins, expires, readied players are locked as roundstart players, role alloc, and everyone confirming they're ready with their character generated or picked, or another timer expires (and the player loses their roundstart slot? they are forced into their last-selected char, a random one? do you go pass a role on if a player drops during the holding phase?) that starts bringing up how round start is currently designed in general.

The most acute worry here is that if you go end of round timer -> kick to lobby -> role select timer -> role selection -> character select timer -> spawn, and you tell someone they're a tot or a nukie at the start, it's a design that clearly encourages rollers. You've not even spawned yet, you can't affect the round by going SSD, so just DC after you get atmos tech when you wanted nukie and hit the next server.

There's also some stuff about tot that comes up if you throw "you're a tot" at people at T=0 in the round, so you'd need to consider how you relay objectives to tots/give them access to the TC store once they've been en-tot-ened.

**Comment by metalgearsloth** [comment]
I'm mostly interested in the decoupling jobs from characters side so we can remove stuff like Loadout names.

**Comment by FairlySadPanda** [comment]
Yeah, it's just this overall roundstart flow chart stuff to decide on. If you go roles-have-characters rather than characters-have-roles you can keep the current system, but I think you need to then mandate the server picks someone's char from their role-characters set (or generates them a random char?) if you want to avoid creating a new step in the roundstart flow?

**Comment by Quantum-cross** [comment]
This is kind of fixed/implemented by #36493
